tools/envoy_collect/README.md

```diff
@@ -0,0 +1,36 @@
+# Collect Envoy stats/log/profile.
+
+The `envoy_collect.py` wrapper script supports gathering up a bundle of useful artifacts from Envoy's
+execution for the purpose of performance profiling or debugging.
+
+## Debugging
+
+To collect verbose logs, stats and other accessible data from the admin endpoint in a tarball,
+invoke `/path/to/envoy-static` as follows under the `envoy_collect.py` wrapper:
+
+```
+envoy_collect.py --envoy-binary /path/to/envoy-static --output-path /path/to/debug.tar -c \
+  /path/to/envoy-config.json <other Envoy args...>
+```
+
+Envoy will run as normal, but when interrupted by SIGINT (e.g. via `kill -s INT` or ctrl-c on
+stdin), it will write to `/path/to/debug.tar` a dump of various logs and the admin endpoint
+handlers. The wrapper configures Envoy for maximum logging verbosity.
+
+This tarball may be useful to attach to issues when reporting. However, a high degree of caution
+is recommended here, as the logs are verbose and will reveal low level traffic details. It is 
+NOT recommended to attach a tarball with any real traffic to an Envoy issue, as it will be placed
```

Comment:
nit: might rephrase slightly in the sense that it's not recommended if there are any privacy concerns whatsoever, otherwise the data should be manually sanitized prior to posting.